osu.Server.Spectator/Hubs/Multiplayer/MultiplayerHub.cs

```diff
@@ -233,6 +233,46 @@ public async Task LeaveRoom()
             }
         }
 
+        public async Task InvitePlayer(int userId)
+        {
+            using (var db = databaseFactory.GetInstance())
+            {
+                bool isRestricted = await db.IsUserRestrictedAsync(userId);
+                if (isRestricted)
+                    throw new InvalidStateException("Can't invite a restricted user to a room.");
+
+                var relation = await db.GetUserRelation(CurrentContextUserId, userId);
+                if (relation?.foe ?? false)
+                    throw new UserBlockedException();
+
+                var inverseRelation = await db.GetUserRelation(userId, CurrentContextUserId);
+
+                if (inverseRelation == null)
+                {
+                    bool userAllowsPMs = await db.GetUserAllowsPMs(userId);
+                    if (!userAllowsPMs)
+                        throw new UserBlocksPMsException();
+                }
+                else if (inverseRelation.foe)
+                    throw new UserBlockedException();
```

Comment:
Not sure about the same exception being used for blocking in both directions. Maybe okay?